cocos/scripting/lua-bindings/manual/Cocos2dxLuaLoader.cpp

```diff
@@ -77,33 +73,55 @@ extern "C"
         {
             if (next == std::string::npos)
                 next = searchpath.length();
-            std::string prefix = searchpath.substr(begin, next);
+            std::string prefix = searchpath.substr(begin, next-begin);
             if (prefix[0] == '.' && prefix[1] == '/')
-            {
                 prefix = prefix.substr(2);
-            }
 
-            pos = prefix.find("?.lua");
-            chunkName = prefix.substr(0, pos) + filename + BYTECODE_FILE_EXT;
-            if (utils->isFileExist(chunkName))
+            pos = prefix.rfind(BYTECODE_FILE_EXT);
+            if (pos != std::string::npos && pos == prefix.length() - BYTECODE_FILE_EXT.length())
+            {
+                prefix = prefix.substr(0, pos);
+            }
+            else
+            {
+                pos = prefix.rfind(NOT_BYTECODE_FILE_EXT);
+                if (pos != std::string::npos && pos == prefix.length() - NOT_BYTECODE_FILE_EXT.length())
+                    prefix = prefix.substr(0, pos);
+            }
+            pos = prefix.find_first_of("?", 0);
+            while (pos != std::string::npos)
+            {
+                prefix.replace(pos, 1, filename);
+                pos = prefix.find_first_of("?", pos + filename.length() + 1);
+            }
+            chunkName = prefix + BYTECODE_FILE_EXT;
+            if (utils->isFileExist(chunkName) && !utils->isDirectoryExist(chunkName))
             {
                 chunk = utils->getDataFromFile(chunkName);
                 break;
             }
             else
             {
-                chunkName = prefix.substr(0, pos) + filename + NOT_BYTECODE_FILE_EXT;
-                if (utils->isFileExist(chunkName))
+                chunkName = prefix + NOT_BYTECODE_FILE_EXT;
+                if (utils->isFileExist(chunkName) && !utils->isDirectoryExist(chunkName))
                 {
                     chunk = utils->getDataFromFile(chunkName);
                     break;
                 }
+                else
+                {
+                    chunkName = prefix;
```

Comment:
Someone may create no-extension file or file with non-standart extension and try to `require` it.

https://www.lua.org/manual/5.1/manual.html#pdf-package.loaders provides an example: `package.path` contain `./?.lc`.

Maybe, somone want to prevent compilation into bytecode. Anyway, it acts like reference says.